### ISSUE TITLE
Crashing when swiping through Gallery

### ISSUE THREAD

**Issue by GetNumber** [avluis/Hentoid#658]
**Hentoid Version**: 1.12.7 (257)

**Device/Android Version**: Fire HD 10 2017 (suez) / LineageOS 12.1-20200505

**Issue details / Repro steps**: When opening downloaded Content, while swiping through the Gallery upon loading a new Image the Screen stays black and the image does not load.
I Fixed the Issue Codewise by adding "android:largeHeap="true"" to the "AndroidManifest.xml" and recompiling.

**Stack trace / LogCat**:
```10-05 09:50:38.282 28916-28964/me.devsaki.hentoid I/art: Alloc concurrent mark sweep GC freed 121(3KB) AllocSpace objects, 1(20KB) LOS objects, 17% free, 65MB/80MB, paused 597us total 47.349ms
10-05 09:50:38.282 28916-28964/me.devsaki.hentoid E/art: Throwing OutOfMemoryError "Failed to allocate a 29196812 byte allocation with 14995536 free bytes and 14MB until OOM"
10-05 09:50:38.282 28916-28964/me.devsaki.hentoid D/skia: --- decoder->decode returned false
```

@AVnetWS/admin-team

**Comment by RobbWatershed** [comment]
Not sure we can fix OOM issues that easily. There are no known leaks on the Gallery screen, and memory management is done as tightly as we can, given our current knowledge.

@GetNumber would you like your fix (`android:largeHeap="true"`) to be included in the regular release of Hentoid ?

**Comment by GetNumber** [comment]
From my understanding the error does happen because the device is always low on free memory and image data in ram can temporary increase in size while working with BitmapFactory class (especially when transformation or scaling is done). 
I'm not an Android-Developer (C++/C# Guy) so my understanding of the Problem may be lacking. 

@RobbWatershed I would be happy if you included the fix into the official release. But please take into consideration that I'm not sure if it may introduce a performance hit on other devices that would not need this fix/workaround.

**Comment by RobbWatershed** [comment]
`BitmapFactory` and large image transformation sure play an important role, but they are not the only part :
- the app handles its downloads with Volley, which is fast but has the downside of keeping downloaded data in memory until the download is finished
- when zooming on very large images, our viewer caches low-res versions of the image to speed up unzooming without having to read the bitmap a 2nd time. That might be optimized a little better

Anyway, PR is accepted. Thanks a lot for bringing that up ! 👍

**Comment by RobbWatershed** [comment]
FYI, this fix has been published on today's v1.12.9
